crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -683,6 +693,79 @@ impl EventCache {
             }
         }
     }
+
+    #[cfg(feature = "experimental-search")]
+    fn parse_timeline_event(event: &TimelineEvent) -> Option<AnySyncMessageLikeEvent> {
+        use ruma::events::AnySyncTimelineEvent;
+
+        if event.kind.is_utd() {
+            return None;
+        }
+
+        match event.raw().deserialize() {
+            Ok(event) => match event {
+                AnySyncTimelineEvent::MessageLike(event) => Some(event),
+                AnySyncTimelineEvent::State(_) => None,
+            },
+
+            Err(e) => {
+                warn!("failed to parse event: {e:?}");
+                None
+            }
+        }
+    }
+
+    /// Takes a [`TimelineEvent`] and passes it to the [`RoomIndex`] of the
+    /// given room which will add/remove/edit an event in the index based on
+    /// the event type.
+    #[cfg(feature = "experimental-search")]
+    async fn search_indexing_task(
```

Comment:
Can you add an `#[instrument(skip_all)]` line on top of this, so the log lines would be tagged accordingly?